### DISCUSSION TITLE
Vehicle models are not tracked at all horizontally

### DISCUSSION THREAD

**Discussion by Tianfneg** [TinyMPC/TinyMPC#29] [Q&A]
Hello, the top three figures in each grid are the desired trajectories I was given, and the bottom three figures are the output from our tracking process. Is
![pig](https://github.com/TinyMPC/TinyMPC/assets/147623854/252e5ccb-eaa7-428c-b8f3-faa0f2c9defa)
 it because I entered the wrong type of reference trajectory?

**Comment by xkhainguyen** [comment]
Hi there, could you please describe your problem setup? What are you trying to do?

**Reply by Tianfneg** [reply]
> 您好，您能描述一下您的问题设置吗？你想做什么？

![image](https://github.com/TinyMPC/TinyMPC/assets/147623854/12cac61e-d5ba-47e5-bfc4-b615529c2402)
Here's my status and input. The effect of my tracking after a given trajectory is as above. There is basically no effect on the last two items above the state parameter. I'm trying to understand if the parameters I'm given should be for an absolute coordinate system or a vehicle coordinate system. I'm in the absolute coordinate system.

**Reply by xkhainguyen** [reply]
Hi, it seems like you are using a car-like nonlinear lateral dynamics. What state are you linearizing this about? I doubt that this TinyMPC implementation will work for you since it only handle LTI systems -- your dynamics is hard to linearizing at one single state in the absolute coordinate system. There might also be tricks to make it LTI in some other coordinate systems. 

You can refer to [this repo](https://github.com/RoboticExplorationLab/TinyMPC-AL) if you want to solve nonlinear bicycle dynamics but this previous effort is not fast and efficient.
